test/refresh.spec.js

```diff
@@ -0,0 +1,73 @@
+import utils from "./config/utils.js";
+import Userfront from "../src/index.js";
+
+import { getIframe, triageEvent } from "../src/iframe.js";
+import { refresh } from "../src/refresh.js";
+import { setCookiesAndTokens } from "../src/cookies.js";
+
+jest.mock("../src/cookies", () => {
+  return {
+    __esModule: true,
+    setCookiesAndTokens: jest.fn(),
+  };
+});
```

Comment:
This works and is great for mocking an entire module, but is problematic when you want one of the functions to not be mocked. Another thing is that you can't return outside variables in the implementation of a mocked method. For example:

```js
const mockResult = {
  prop: "val",
};

jest.mock("../src/cookies", () => {
  return {
    __esModule: true,
    setCookiesAndTokens: jest.fn(() => {
      return mockResult;
    }),
  };
});
```

Will throw the following error: `The module factory of jest.mock() is not allowed to reference any out-of-scope variables. Invalid variable access: _get__`.
The docs say that this should work if the variable you're using begins with 'mock', but I haven't gotten it to work properly even after copy/pasting [their example](https://jestjs.io/docs/es6-class-mocks#calling-jestmock-with-the-module-factory-parameter). There are ways around it, such as using `mockImplementation()` inside a test, so that's not a big issue but I figured I'd mention it.

Reply:
We'll have to see it in practice, but it feels like a bit of an anti-pattern if you're only mocking parts of a file.  Generally if we want to mock something it means we aren't testing that part of the codebase.